src/topology/sheaves/Godement.lean

```diff
@@ -0,0 +1,124 @@
+
+section presheaf
+
+open Top
+open topological_space
+open category_theory
+open category_theory.limits
+
+universes u v
+
+variables {X : Top.{u}} {C : Type u} [category.{u} C]
+variables [has_limits C] [has_terminal C] [has_colimits C]
+variables [Π (x : X) (U : opens X), decidable (x ∈ U)]
+variables (𝓕 : presheaf C X)
+
+/--
+The `godement_presheaf` for a presheaf `𝓕` is defined as a product presheaf `∏ₓ skyscraper(𝓕ₓ)`
+-/
+def godement_presheaf : presheaf C X :=
+∏ (λ x, skyscraper_presheaf x (𝓕.stalk x) : X → presheaf C X)
+
+/--
+The sections of `godement_presheaf` on opens `U` is isomorphic to `∏ₓ skyscraper(x, 𝓕ₓ)(U)`, i.e.
+the categorical definition and the concrete definition agree.
+-/
+@[simps] def godement_presheaf_obj (U : (opens X)ᵒᵖ) :
+  (godement_presheaf 𝓕).obj U ≅ ∏ (λ x, (skyscraper_presheaf x (𝓕.stalk x)).obj U) :=
+limit_obj_iso_limit_comp_evaluation _ _ ≪≫
+{ hom := lim_map { app := λ _, 𝟙 _, naturality' := by { rintros ⟨x⟩ ⟨y⟩ ⟨⟨(rfl : x = y)⟩⟩, refl } },
+  inv := lim_map { app := λ _, 𝟙 _, naturality' := by { rintros ⟨x⟩ ⟨y⟩ ⟨⟨(rfl : x = y)⟩⟩, refl } },
+  hom_inv_id' :=
+  begin
+    ext,
+    erw [category.assoc, lim_map_π, ←category.assoc, lim_map_π, category.id_comp, category.comp_id,
+      category.comp_id],
+  end,
+  inv_hom_id' :=
+  begin
+    dsimp,
+    ext,
+    erw [category.assoc, lim_map_π, ←category.assoc, lim_map_π, category.comp_id, category.id_comp,
+      category.comp_id],
+  end }
+
+/--
+Let `U` be an open set, since `𝓕(U) ⟶ 𝓕ₓ` or `𝓕(U) ⟶ *` depending on `x ∈ U` or not where `*`
+is a terminal object, there is a product map `𝓕(U) ⟶ ∏ₓ, 𝓕ₓ or *`.
+-/
+def to_godement_presheaf_aux (U : (opens X)ᵒᵖ) :
+  𝓕.obj U ⟶ ∏ λ (x : X), (skyscraper_presheaf x (𝓕.stalk x)).obj U :=
+pi.lift $ λ x, if m : x ∈ U.unop
+  then 𝓕.germ ⟨x, m⟩ ≫ eq_to_hom (by rw [skyscraper_presheaf_obj, if_pos m, subtype.coe_mk])
+  else terminal.from _ ≫ eq_to_hom (by rw [skyscraper_presheaf_obj, if_neg m])
+
+/--
+Let `U` be an open set, if `p ∈ U`, then there is morphism `𝓕(U) ⟶ 𝓕ₚ` by composing the product
+map `to_godement_presheaf_aux` with projection map `pi.π`. This agrees with the `germ` morphism.
+-/
+def to_godement_presheaf_aux_comp_π {U : (opens X)ᵒᵖ} (p : U.unop) :
+  𝓕.obj U ⟶ 𝓕.stalk p :=
+to_godement_presheaf_aux 𝓕 U ≫ pi.π _ p ≫ eq_to_hom (if_pos p.2)
+
+@[simp] lemma to_godement_presheaf_aux_comp_π_eq {U : (opens X)ᵒᵖ} (p : U.unop) :
+  to_godement_presheaf_aux_comp_π 𝓕 p = presheaf.germ 𝓕 p :=
+begin
+  dunfold to_godement_presheaf_aux_comp_π presheaf.germ to_godement_presheaf_aux,
+  rw [←category.assoc, limit.lift_π],
+  simp only [fan.mk_π_app],
+  split_ifs,
+  { rw [category.assoc, eq_to_hom_trans, eq_to_hom_refl, category.comp_id],
+    refl },
+  { exfalso, exact h p.2, },
+end
+
+/--
+Under the isomorphism `godement_presheaf(𝓕, U) ≅ ∏ₓ skyscraper(x, 𝓕ₓ)(U)`, there is a morphism
+`𝓕 ⟶ ∏ₓ skyscraper(x, 𝓕ₓ) ≅ godement_presheaf(𝓕)`
+-/
+@[simps] def to_godement_presheaf : 𝓕 ⟶ godement_presheaf 𝓕 :=
```

Comment:
Would it be easier to work with the definition
```
def to_godement_presheaf : 𝓕 ⟶ godement_presheaf 𝓕 :=
pi.lift $ λ p₀, (stalk_skyscraper_presheaf_adjunction_auxs.unit p₀).app 𝓕
```
? I guess it depends on what you want to do with it?

Reply:
This definition is certainly more concise. All I need from this definition is [this lemma](https://github.com/leanprover-community/mathlib/blob/28dc545db40e29caa3bf6a1db545083e055f40df/src/topology/sheaves/godement.lean#L153), which should also hold for your definition modulo some modification. So I will use your definition instead.

Reply:
I mentioned this to you [a while ago](https://github.com/leanprover-community/mathlib/pull/15742#discussion_r951803305) but maybe you didn't notice :)